library/rsa.c

```diff
@@ -1088,29 +1098,68 @@ int mbedtls_rsa_private( mbedtls_rsa_context *ctx,
 }
 
 #if defined(MBEDTLS_PKCS1_V21)
+#if !defined(MBEDTLS_MD_C)
+static int ret_from_status( psa_status_t status )
```

Comment:
This function has future potential to be refactored to `psa_to_mbedtls_error( psa_status_t status, int module )` to have it uniform across modules.

Reply:
Good point! Could come in handy for #6146 and #6147. What would be a good home for this function? Would `library/hash_info.h` work? (Since it's specific to MD.)

Btw, it should probably be renamed `mbedtls_md_error_from_psa()` since it translates to MD error, and for consistency for our naming scheme (the `psa_` namespace is not for internal things) and the existing `mbedtls_pk_error_from_psa()`.

Reply:
I was thinking about using it globally across Mbed TLS, hence the `int module` argument - this way there would some module-specific errors first (in a `switch( module )`, followed by a fallback - presumably, set of shared error codes that are the same for every module.

Reply:
Oh, sorry, I read too fast and missed that part. Yes, why not, it there's some code that can be shared between modules, we could save some code size that way. But I'm slightly worried that the function would become big and hard to read. Also, sometimes even within a module you need variants, see `mbedtls_pk_error_from_psa_rsa()` which may return either RSA or PK error codes.

I'm inclined to keep this focused on MD as a first step, then look into consolidation with other modules only later, unless you feel strongly otherwise.

Reply:
Absolutely no strong feelings, it's just a thing that came to my mind, as I can see that we're duplicating similar code in a couple of places.

Reply:
Agreed. Whoever adopts this PR, I suggest moving this function to `library/hash_info.[ch]` under the name `mbedtls_md_error_from_psa()`. That way, #6146, #6147 and #6149 can use it too. It doesn't solve the problem of cross-module near-duplication, but at least it will avoid exact-duplication for all cases where we silently fall back to PSA in the absence of MD.

Regarding the larger picture, @AndrzejKurek can you create an issue about that so we don't forget about it? I guess the "Use PSA: misc. gaps" EPIC would be a good place.

Reply:
Renamed function `ret_from_status->mbedtls_md_error_from_psa` and moved to library/hash_info.[ch] as global function.

Reply:
Issue added here: https://github.com/Mbed-TLS/mbedtls/issues/6158 with a remark to start by assesing if the potential gain is worth it.